File: movinProject/src/main/java/com/example/movinProject/main/debateRoom/dto/DebateRoomDto.java
package com.example.movinProject.main.debateRoom.dto;

import com.example.movinProject.main.movie.dto.MovieDto;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

@Getter
@Builder
public class DebateRoomDto {
    private Long id;
    private String title;
    private String topic;
    private String state;
    private MovieDto movie;

    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", timezone = "Asia/Seoul")
    private LocalDateTime startTime;

    private int duration;
    private int maxUserNumber;
    private int agreeJoinedUserNumber;
    private int disagreeJoinedUserNumber;
}

File: src/models/dto/DebateRoomDto.ts
import ChatDto from './ChatDto';
import { SimpleMovieDto } from './MovieDto';

export interface DebateRoomDto {
  id: number;
  title: string;
  topic: string;
  state: 'OPEN' | 'DISCUSS' | 'VOTE' | 'CLOSE';
  movie: SimpleMovieDto;
  startTime: string;
  duration: number;
  maxUserNumber: number;
  agreeJoinedUserNumber: number;
  disagreeJoinedUserNumber: number;
}

export interface DebateRoomVoteDto {
  id: number;
  voted: boolean;
  voteAgree: boolean;
  joined: boolean;
  agree: boolean;
  movie: SimpleMovieDto;
  title: string;
  topic: string;
  stateType: 'OPEN' | 'DISCUSS' | 'VOTE' | 'CLOSE';
  startTime: string;
  duration: number;
  maxUserNumber: number;
  agreeJoinedUserNumber: number;
  disagreeJoinedUserNumber: number;
}

export interface DebateRoomChatDto {
  id: number;
  voted: boolean;
  voteAgree: boolean;
  joined: boolean;
  agree: boolean;
  movie: SimpleMovieDto;
  title: string;
  topic: string;
  stateType: 'OPEN' | 'DISCUSS' | 'VOTE' | 'CLOSE';
  startTime: string;
  duration: number;
  maxUserNumber: number;
  agreeJoinedUserNumber: number;
  disagreeJoinedUserNumber: number;
  summarize: string | null;
  chats: ChatDto[];
}